.github/workflows/windows.yml

```diff
@@ -138,11 +138,12 @@ jobs:
 
       - name: pip-packages
         run: |
-          python -m pip install build delvewheel numpy cython pybind11 meson-python meson ninja pytest pytest-xdist pytest-timeout pooch hypothesis
+          python -m pip install build delvewheel cython pybind11 meson-python meson ninja pytest pytest-xdist pytest-timeout pooch hypothesis
+          python -m pip install --pre --upgrade --timeout=60 -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy
 
       - name: Build
         run: |
-          python -m build -Csetup-args="-Duse-pythran=false"
+          python -m build --no-isolation -x -Csetup-args="-Duse-pythran=false"
```

Comment:
> Ok, I guess my point is that I'm probably not causing the problems with my diff.

Actually, it's quite likely that you _are_ causing this with this change; I only understood this relationship now after reading some more of the pytest issue.

Reply:
I don't follow, your comment is on a diff for a CI job that is passing, and has been passing for some time in this PR.

Reply:
Two CI jobs here are failing (though in jobs that aren't always run, so I don't know if they fail without the diff here; tried and failed to check when they were last run, and if pytest 8 was present then).

Both failing jobs are pulling in pytest 8.0.0rc1, and the pytest issue [describes](https://github.com/pytest-dev/pytest/issues/9765#issuecomment-1869103766) that the problem is related to not using isolation, which this PR does. These are the ingredients for my suspicion.

In the pytest issue you [mention](https://github.com/pytest-dev/pytest/issues/9765#issuecomment-1879885107) that the job without isolation is passing, but perhaps it's not using the pytest rc? Also, my suspicion would be easy to test by simply removing `--no-isolation` here and seeing if the wheel builds pass.

The alternative would be to simply restrict to `pytest <8` in the wheel builds now.

Reply:
Ah, I get your point now; the wheel builds are not built by this workflow, right? I got tripped up by the naming, as `.github/workflows/windows.yml` is pretty generic.

Reply:
Right, I didn't touch the wheel builds in this PR.

To be fair, you aren't entirely wrong to suspect disabled build isolation, it should be off in the wheel builds too, because we are intentionally pulling in the NumPy pre-release, both in the wheel builds and in the CI job that I modified at a reviewer's request here.

As for your ideas, yeah you'll see above that I say:

> ... I'm also not convinced we should ever be using pytest release candidates in our wheel builds like here (pytest-8.0.0rc1).

Perhaps I'll try pushing that in, that I'm not sure it should really be needed to move forward with fixing the few hundred test failures here.

It isn't clear to me that build isolation itself is involved, but the NumPy pre-release with Python versions below `3.11` seems to be a common issue.